cmd/spire-server/cli/validate/validate.go

```diff
@@ -32,11 +39,38 @@ func (c *validateCommand) Synopsis() string {
 }
 
 func (c *validateCommand) Run(args []string) int {
-	if _, err := run.LoadConfig(commandName, args, nil, c.env.Stderr, false); err != nil {
+	config, err := run.LoadConfig(commandName, args, []log.Option{log.WithOutputFile(os.DevNull)}, c.env.Stderr, false)
+
+	if err != nil {
 		// Ignore error since a failure to write to stderr cannot very well be reported
 		_ = c.env.ErrPrintf("SPIRE server configuration file is invalid: %v\n", err)
 		return 1
 	}
-	_ = c.env.Println("SPIRE server configuration file is valid.")
-	return 0
+
+	s := server.New(*config)
+
+	pluginNotes, err := s.ValidateConfig(context.Background())
+	if err != nil && status.Code(err) != codes.Unimplemented {
+		_ = c.env.ErrPrintf("Could not validate configuration file: %v", err)
+		return 1
+	}
+
+	if len(pluginNotes) == 0 {
```

Comment:
it may be a little confusing, here we are:
1-> error if error exists and retun 1
2-> if no error and notes success retunr 0
3-> if notes > 0 print errors  return 1

What do you think about move 3 to 2, so we have success at the end, 
or even create a function so we can reduce the amount of return 1